srv/salt/_modules/cephdisks.py

```diff
@@ -2,82 +2,429 @@
+    def hw_raid_ctrl_detection(self):
+        """
+        Calls out for lspci to retrieve information
+        about the underlying RAID-Controller
+        return:
+           (dict): Information about RAID
+        """
+        info = {}
+        lspci_path = self.which('lspci')
+        cmd = "{} -vv | grep -i raid".format(lspci_path)
+        proc = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+        for line in proc.stdout:
+            """
+            match one of the available raid ctrls
+            areca, megaraid, 3ware, hprr
+            """
+            if 'megaraid' in line.lower():
+                info['controller_name'] = 'megaraid'
+            elif 'areca' in line.lower() or 'arcmsr' in line.lower():
+                info['controller_name'] = 'areca'
+            elif '3ware' in line.lower():
+                info['controller_name'] = '3ware'
+            elif 'hprr' in line.lower():
+                info['controller_name'] = 'hprr'
+            elif 'hpt' in line.lower():
+                info['controller_name'] = 'hpt'
+            elif 'cciss' in line.lower():
+                info['controller_name'] = 'cciss'
+            elif 'aacraid' in line.lower():
+                info['controller_name'] = 'aacraid'
+            else:
+                info['controller_name'] = None
+
+            if info['controller_name']:
+                info['raidtype'] = 'hardware'
+                msg = 'Found raidctrl: {}'.format(info['controller_name'])
+                log.info(msg)
+                return info
+        if not info['controller_name']:
+            info['raidtype'] = None
+            log.info("No raidctrl found")
+            return info
+
+    def which(self, program, nofail=True):
+        """
+        Instead of using python's built-in _platform_ we rely
+        on the tools presence.
+
+        Sidenote for testing: smartctl, hwinfo and lshw resides in /sbin/
+        Detection will only work in privileged environments
+        args:
+            programm(str): name of programm you want to retrieve the path from
+        return:
+            str: the full path of the programm
+        """
+        def is_exe(fpath):
+            return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+
+        fpath, fname = os.path.split(program)
+        if fpath:
+            if is_exe(program):
+                return program
+        else:
+            for path in os.environ["PATH"].split(os.pathsep):
+                path = path.strip('"')
+                exe_file = os.path.join(path, program)
+                if is_exe(exe_file):
+                    return exe_file
+        if nofail is True:
+            return None
+        elif nofail is False:
+            raise StandardError("Can't find the tool: {}. Please Install it in order to resume.".format(program))
+        else:
+            raise StandardError("nofail needs to be bool(True) or bool(False)")
+        
```

Comment:
There's a bunch of unnecessary whitespaces here. I'm always a fan of removing these.